lib/withCache.ts

```diff
@@ -0,0 +1,49 @@
+import { NextApiRequest, NextApiResponse } from 'next';
+import { CronJob } from 'cron';
+import cache from './cache';
+import { fetchHakkasanByRef } from './hakkasan';
+import type { HakkasanByRefResponse } from './hakkasan';
+import type { Cache } from './cache';
+
+export interface HakkasanRequest extends NextApiRequest {
+  query: {
+    ref: string | string[];
+  };
+  cache: Cache<HakkasanByRefResponse>;
+}
+
+type Handler = (req: HakkasanRequest, res: NextApiResponse) => Promise<void>;
+
+const cron = new CronJob(
+  '00 00 00 * * *',
+  // '* * * * * *', // Cron job every second for testing purposes
```

Comment:
This expression tells your cron to run every minute. Cron only allows for a minimum of one minute!

Reply:
I remember seeing that in the docs, but the `cron` repo's `examples` folder has this "Every second" example: https://github.com/kelektiv/node-cron/blob/master/examples/basic.js. 🤔

Reply:
Oh interesting, my bad I didn't notice that six asterisks were being used! Looks like `node-cron` supports a non-standard extra value to indicate seconds. I'll resolve this :)